version: 1.0

description: A workflow demonstrating output on error.

vars:
  - progress: 0

output:
  - progress: <% ctx().progress %>

tasks:
  task1:
    action: core.noop
    next:
      - when: <% succeeded() %>
        publish:
          - progress: 25
        do:
          - task2
  task2:
    action: core.local cmd="exit 1"
    next:
      - when: <% succeeded() %>
        publish:
          - progress: 50
        do:
          - task3
  task3:
    action: core.noop
    next:
      - when: <% succeeded() %>
        publish:
          - progress: 75
        do:
          - task4
  task4:
    action: core.noop
    next:
      - when: <% succeeded() %>
        publish:
          - progress: 100
